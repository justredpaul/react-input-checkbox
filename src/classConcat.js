export const classConcat = classes => {
  let type = typeof classes;
  if (type === "string" || type === "number") return classes || "";

  if (Array.isArray(classes) && classes.length > 0) {
    return classes.reduce((concated, className) => {
      return className ? concated.concat(`${className} `) : concated;
    }, "");
  } else {
    return Object.keys(classes).reduce((concated, className) => {
      return classes[className] ? concated.concat(`${className} `) : concated;
    }, "");
  }
};
