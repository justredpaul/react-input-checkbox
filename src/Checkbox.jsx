import React from 'react';

import { classConcat } from './classConcat';

import './style.css';

export const Checkbox = props => (
  <label className={classConcat({
    'Checkbox_themed': props.theme,
    [props.theme]: props.theme
  })}>
    <input type="checkbox"
      className={classConcat({
        'Checkbox__input_themed': props.theme,
        'Checkbox__input_themed_indeterminate': props.indeterminate,
        [`${props.theme}__input`]: props.theme,
        [`${props.theme}__input_indeterminate`]: props.indeterminate
      })}
      disabled={props.disabled}
      checked={props.value}
      onChange={props.onChange} />
    {props.theme && (
      <span className={classConcat([
        'Checkbox__image_themed',
        `${props.theme}__image`
      ])} />
    )}
    <span className={classConcat({
      [`${props.theme}__label`]: props.theme
    })}>{props.children}</span>
  </label>
);
