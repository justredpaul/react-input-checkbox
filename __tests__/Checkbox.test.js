import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { Checkbox } from '../src/Checkbox';

describe('Checkbox', () => {
  // Arrange
  it('should render checkbox', () => {
    const { container } = render(<Checkbox />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render checkbox label from children', () => {
    const { container, getByText } = render(<Checkbox>Some text for label</Checkbox>);

    expect(getByText('Some text for label')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render checkbox image if theme provided', () => {
    const { container } = render(<Checkbox theme="test-checkbox">Label</Checkbox>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render class names if theme provided', () => {
    const { container } = render(<Checkbox theme="test-checkbox">Label</Checkbox>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
