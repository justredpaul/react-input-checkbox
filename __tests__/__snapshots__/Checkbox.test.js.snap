// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render checkbox 1`] = `
<label
  class=""
>
  <input
    class=""
    type="checkbox"
  />
  <span
    class=""
  />
</label>
`;

exports[`Checkbox should render checkbox image if theme provided 1`] = `
<label
  class="Checkbox_themed test-checkbox "
>
  <input
    class="Checkbox__input_themed test-checkbox__input "
    type="checkbox"
  />
  <span
    class="Checkbox__image_themed test-checkbox__image "
  />
  <span
    class="test-checkbox__label "
  >
    Label
  </span>
</label>
`;

exports[`Checkbox should render checkbox label from children 1`] = `
<label
  class=""
>
  <input
    class=""
    type="checkbox"
  />
  <span
    class=""
  >
    Some text for label
  </span>
</label>
`;

exports[`Checkbox should render class names if theme provided 1`] = `
<label
  class="Checkbox_themed test-checkbox "
>
  <input
    class="Checkbox__input_themed test-checkbox__input "
    type="checkbox"
  />
  <span
    class="Checkbox__image_themed test-checkbox__image "
  />
  <span
    class="test-checkbox__label "
  >
    Label
  </span>
</label>
`;
